bst.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+import random
+import subprocess
+
+
+class Bst(object):
+    """Binary Search Tree using a dictionary"""
+    def __init__(self, value=None):
+        self.tree = {}
+        self.top = None
+        self._size = 0
+        self._depth = 0
+
+        if value is not None:
+            self.tree[value] = {'depth': 1}
+            self.top = value
+            self._size += 1
+            self._depth += 1
+
+    def insert(self, value):
+        """Insert a node with value in order.
+
+        In value is already present, it is ignored."""
+        current = self.top
+        if current is None:
+            self.tree[value] = {'depth': 1}
+            self.top = value
+            self._size += 1
+            self._depth += 1
+            return
+        depth = 1  # starts trying to insert at depth 2
+        while current is not None:
+            depth += 1
+            if current == value:
+                return
+            if current < value:
+                traverse = self.tree[current].get('right')
+                child = 'right'
+            else:
+                traverse = self.tree[current].get('left')
+                child = 'left'
+            if traverse is None:
+                #actual insert
+                self.tree[value] = {'depth': depth}
+                self.tree[current][child] = value
+                self._size += 1
+                if depth > self._depth:
+                    self._depth = depth
+                return
+            current = traverse
```

Comment:
Given the upcoming need to traverse, you might consider breaking the code responsible for determining where to traverse to into a support method.